Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -226,6 +227,14 @@ namespace AZ::DocumentPropertyEditor
             m_onChangedCallbacks.SetValue(m_builder.GetCurrentPath(), AZStd::move(onChanged));
             m_builder.AddMessageHandler(m_adapter, Nodes::PropertyEditor::OnChanged);
             m_builder.AddMessageHandler(m_adapter, Nodes::PropertyEditor::RequestTreeUpdate);
+
+            if (hashValue)
+            {
+                AZStd::any anyVal(&instance);
+                m_builder.Attribute(
+                    Nodes::PropertyEditor::ValueHashed,
+                    AZ::Uuid::CreateData(reinterpret_cast<AZStd::byte*>(AZStd::any_cast<void>(&anyVal)), anyVal.get_type_info().m_valueSize));
+            }
```

Comment:
FYI @dtamkin1 @alexmontAmazon This will only hash the value type stored within the AZStd::any.
It will not be able to punch through to introspect pointer types.
If a structure has a pointer whose address value hasn't changing, but what the pointer points to changes, then this will hash to the same value
```c++
struct Bar
{
    int m_value{};
};
struct Foo
{
    Bar* m_pointer{};
};

Foo f1;
f1.m_pointer = new Bar{ 1 };

// Only hashes the 8 byte pointer value stored in Foo
auto oldValue = AZ::Uuid::CreateData(reinterpret_cast<AZStd::byte*>(&Foo), sizeof(Foo));
// Update pointer value to 2
f1.m_pointer.m_value = 2;
// rehash
auto newValue = AZ::Uuid::CreateData(reinterpret_cast<AZStd::byte*>(&Foo), sizeof(Foo));
// Both the old and new uuid values are the same
delete f1.m_pointer;
```

Reply:
@lumberyard-employee-dm  noted. Can we look into hashing the (assumedly opaque) type that the pointer is referencing instead of, or in addition to, the pointer value? I imagine that we really want both, or if we had to settle for one, it'd be the hash of the actual object.

Reply:
Well the problem is, we don't know what pointer types the opaque type is referencing.
Using the approach of storing a full copy of the opaque type and then comparing it later using `AZ::Helpers::CompareAnyValue` would work.

But in this case, let's say we hashed an `AZ::Entity` value type.
Well the AZ::Entity class has a vector of `AZ::Component*` [on it](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/Entity.h#L407-L408) as well as other members.

The hashing wouldn't take into account any changes to a value within a component on an entity.
Pretty much the comparison aren't "deep".

That probably will not be an issue in 99% of the cases where an object would be stored in an opaque pointer, but could be for a small subset of reflected types being shown though the ReflectionAdapter